application.py

```diff
@@ -80,17 +80,50 @@ def admin():
         return redirect(url_for("register"))
     data = Users.query.order_by(Users.time.desc()).all()
     return render_template("admin.html",data=data)
+
 @app.route("/home")
 def home():
     if session["email"] is None:
         return redirect(url_for("register"))
     message = session["email"]
     return render_template("homepage.html",username=message)
 
+@app.route("/search",methods=["POST","GET"])
+def search():
+    if session["email"] is None:
+        return  redirect(url_for("login"))
+    if request.method == "POST":
+        text = request.form.get("name")
+        option = request.form.get("option")
+        print(text)
+        if option == "title":
+            print(option)
+            books = db.session.execute('SELECT * FROM "Books" WHERE UPPER("title") LIKE UPPER(:text)',{"text":'%'+text+'%'}).fetchall()
+        elif option == "isbn":
+            books = db.session.execute('SELECT * FROM "Books" WHERE UPPER("isbn") LIKE UPPER(:text)',{"text":'%'+text+'%'}).fetchall()
+        elif option == "author":
+            books = db.session.execute('SELECT * FROM "Books" WHERE UPPER("author") LIKE UPPER(:text)',{"text":'%'+text+'%'}).fetchall()
+        else:
+
+            # if text.isnumeric():
```

Comment:
can simplify these queries. You should also implement default search with all the fields when no option is selected

Reply:
I placed "option" as required field. So there is no chance to no option case. Remaining are updated. Would you please check it once.

Reply:
what happens if the user doesn't select the option and clicks search directly

Reply:
<div class="radio" aria-required="true">
                        <input type="radio" name="option" value="title"required>
                        <label for="title">Title</label>
                        <input type="radio" name="option" value="year" required>
                        <label for="Year">Year</label>
                        <input type="radio" name="option" value="isbn" required>
                        <label for="isbn">Isbn</label>
                        <input type="radio" name="option" value="author" required>
                        <label for="author">Author</label>
                    </div>
**As shown in above I placed required in the "option". So every time when user clicks search button directly then app asks him to select any one option. After he selects any option then only searching will be started. Otherwise it still in the same page. I also attached screenshot of it**
![Screenshot (672)](https://user-images.githubusercontent.com/55662250/80539475-317c1700-89c5-11ea-9ccc-310f3730d431.png)

Reply:
ok. there can be any option to search directly with out selecting any of the options, In that case you use OR operation and search in all the fields